src/app/App.js

```diff
@@ -123,6 +123,14 @@ class App extends EventEmitter {
 		 */
 		this.nativeScrollRestorationSupported = ('scrollRestoration' in globals.window.history);
 
+		/**
+		 * When set to true means that the current navigation cannot be Cancelled to start another.
+		 * @type {boolean}
+		 * @default false
+		 * @protected
+		 */
+		this.isNavigationAvoided = false;
```

Comment:
I find this variable a bit confusing... could we try to figure out a different option?

If we stay with boolean vars, other options could be:
- `preventNavigation`
- `scheduleNavigation`
- ...

Another option, maybe more descriptive could be something like `navigationStrategy`, which could be:
- `'immediate'`: starts navigation right away (default value)
- `'schedule'`: queues all navigation events to trigger them in order after the previous one completes (we don't need to implement this option for now)
- `'scheduleLast'`: queues navigation events, keeping only the last one and triggers them when the previous one completes

Reply:
Hey Chema, I liked the navigationStrategy idea.

Reply:
Thats good too! I'll take a look at this!